File: News_Service_JDBC/src/main/java/com/dunice/projectNews/controllers/FileController.java
package com.dunice.projectNews.controllers;

import com.dunice.projectNews.dto.common.CustomSuccessResponse;
import com.dunice.projectNews.services.FileService;
import lombok.RequiredArgsConstructor;
import org.springframework.core.io.UrlResource;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

@RestController
@RequestMapping("/file")
@RequiredArgsConstructor
public class FileController {

    private final FileService fileService;

    @PostMapping("/uploadFile")
    public ResponseEntity<CustomSuccessResponse<String>> uploadFile(@RequestParam(value = "file") MultipartFile file) {
        return ResponseEntity
                .ok()
                .body(new CustomSuccessResponse<>(fileService.uploadFile(file)));
    }

    @GetMapping("{fileName}")
    public ResponseEntity<UrlResource> getFile(
            @PathVariable(value = "fileName") String fileName) {
        return ResponseEntity
                .ok()
                .contentType(MediaType.MULTIPART_FORM_DATA)
                .body(fileService.giveFile(fileName));

    }
}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/controllers/FileController.kt
package com.dunice.advancedServiceKotlin.controllers

import com.dunice.advancedServiceKotlin.dto.common.CustomSuccessResponse
import com.dunice.advancedServiceKotlin.services.FileService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.core.io.UrlResource
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile

@RestController
@RequestMapping("/file")
class FileController @Autowired constructor(private val fileService: FileService) {
    @PostMapping("/uploadFile")
    fun uploadFile(@RequestParam(value = "file") file: MultipartFile?): ResponseEntity<CustomSuccessResponse<String>> {
        return ResponseEntity
            .ok()
            .body(CustomSuccessResponse(fileService.uploadFile(file!!)))
    }

    @GetMapping("{fileName}")
    fun getFile(
        @PathVariable(value = "fileName") fileName: String?
    ): ResponseEntity<UrlResource> {
        return ResponseEntity
            .ok()
            .contentType(MediaType.MULTIPART_FORM_DATA)
            .body(fileService.giveFile(fileName))
    }
}
